hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFileSystemInitialization.java

```diff
@@ -77,6 +77,22 @@ public void testNewInstanceFailure() throws Throwable {
         .isEqualTo(1);
   }
 
+  @Test
+  public void testInvalidLocalFileSystem() throws Exception {
+    Configuration conf = new Configuration();
+    conf.set("fs.file.impl", "org.apache.hadoop.fs.RawLocalFileSystem");
```

Comment:
needs to disable caching for that fs or this can pick up existing one from cache